envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -147,7 +154,12 @@ class SnapshotUpdater(
         }
     }
 
-    private fun updateSnapshotForGroups(groups: Collection<Group>, result: UpdateResult) {
+    private val updateSnapshotForGroupsTimer = meterRegistry.timer("snapshot-updater.update-snapshot-for-groups.time")
```

Comment:
I'd put fields at the top of the class.

Reply:
IMO it is more readable when timer is defined next to the place where it is used.